build.rs

```diff
@@ -84,6 +86,26 @@ fn main() {
     if linux_time_bits64 {
         set_cfg("linux_time_bits64");
     }
+    println!("cargo:rerun-if-env-changed=RUST_LIBC_UNSTABLE_GNU_FILE_OFFSET_BITS64");
+    match env::var("RUST_LIBC_UNSTABLE_GNU_FILE_OFFSET_BITS64") {
+        Ok(val) => {
+            if val == "1" {
+                let target_env = env::var("CARGO_CFG_TARGET_ENV").unwrap();
+                let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
+                let target_ptr_width = env::var("CARGO_CFG_TARGET_POINTER_WIDTH").unwrap();
+                let target_arch = env::var("CARGO_CFG_TARGET_ARCH").unwrap();
+                if target_env == "gnu"
+                    && target_os == "linux"
+                    && target_ptr_width == "32"
+                    && target_arch != "riscv32"
+                    && target_arch != "x86_64"
```

Comment:
`x86_64` won't ever get matched here since `target_ptr_width` is checked right? And then, what is special about riscv32?

It would probably be good to add a riscv32 test in CI since unfortunately we don't have one yet (not here of course)

Reply:
`x86_64-unknown-linux-gnux32` have 32bit pointer but always 64 bit `time_t`. 
`riscv32` also always have 64-bit `time_t`.